ADD: analysis views.py 추가

yj-leee opened the pull request:
## 수정 사항 간략한 한줄 요약
- 별점(0.5점~5점까지)의 개수 카운트 view 추가

## 수정 사항들 자세한 내용
- analysis 앱 StarratingView 추가 / postman 테스트 완료

- movie
영화 상세 페이지에서 각각의 영화에 대해 유저들이 평가한 별점의 개수 리턴 
-users
유저 개인 페이지에서 모든 영화에 대해 평가한 별점의 개수 리턴

## 기타 질문 및 특이 사항
-효율성
movie, user 사용하는 페이지가 다른데 view를 두개로 나누는게 좋을까요?

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

Commit:
ADD: analysis views.py 추가

- analysis 별점 view 추가

devops-andrew commented on users/utils.py:
```diff
@@ -0,0 +1,25 @@
+            access_token  = request.headers['Authorization']
+            payload       = jwt.decode(access_token, settings.SECRET_KEY, algorithms='HS256')
+            user          = User.objects.get(id=payload['id'])
+            request.user  = user.id
```
id를 넘기기보다는 객체를 넘겨주세요.
```suggestion
            request.user  = user
```

devops-andrew commented on analysis/views.py:
```diff
@@ -1,3 +1,34 @@
+class StarratingView(View):
+    @login_decorator
+    def get(self, request):
+        if request.GET.get('id', None) != None:
```
```suggestion
         ???_id  = request.GET.get('id', None) 
        if ???_id:
```
위의 id는 어떤 아이디 일까요? 영화 id 인가요? 
제대로 담기고 None이 아니라면 True 이므로 != None 할 필요 없습니다.

devops-andrew commented on analysis/views.py:
```diff
@@ -1,3 +1,34 @@
+    @login_decorator
+    def get(self, request):
+        if request.GET.get('id', None) != None:
+            movie_id       = request.GET.get('id', None)
```
위에서 받았다면 여기서 또 받을 필요가 없겠죠

devops-andrew commented on analysis/urls.py:
```diff
@@ -0,0 +1,6 @@
+from django.urls import path
+from analysis.views import StarratingView
```
```suggestion
from analysis.views import StarRatingView
```

devops-andrew commented on analysis/urls.py:
```diff
@@ -0,0 +1,6 @@
+from analysis.views import StarratingView
+
+urlpatterns = [
+    path("/my_star", StarratingView.as_view()),
```
```suggestion
    path("/my_star", StarRatingView.as_view()),
```

devops-andrew commented on analysis/views.py:
```diff
@@ -1,3 +1,34 @@
+from movie.models import Movie
+from .models import Star
+
+class StarratingView(View):
```
```suggestion
class StarRatingView(View):
```

devops-andrew reviewed (changes requested):
우선 이부분들 수정 부탁드립니다.

Commit:
MODIFY: analysis views.py 수정

1. 데코레이터 파일 request.user 수정

2. urls.py, views.py Class name 수정

3. view 기능
- movie 상세 페이지의 별점 분포(해당 movie 별점을 체크한 전체 user기준)
- user 마이 페이지의 별점 분포도 (user개인이 체크한 모든 영화에 대한 별점 분포)
- movie_id를 받으면 movie에 대한 별점을 리턴
- movie_id를 받지 않으면 user의 개인 별점 분포 리턴

yj-leee commented:
[](url)
<img width="1113" alt="스크린샷 2020-11-23 오후 11 51 35" src="https://user-images.githubusercontent.com/70866485/99978838-c9ba6d00-2de9-11eb-9633-4331824e5e47.png">
<img width="1140" alt="스크린샷 2020-11-23 오후 11 51 21" src="https://user-images.githubusercontent.com/70866485/99978863-cd4df400-2de9-11eb-96f9-f3fd459523f6.png">
<img width="698" alt="스크린샷 2020-11-23 오후 11 52 03" src="https://user-images.githubusercontent.com/70866485/99978871-d048e480-2de9-11eb-8611-5c10c1a91482.png">
<img width="685" alt="스크린샷 2020-11-23 오후 11 52 38" src="https://user-images.githubusercontent.com/70866485/99978876-d212a800-2de9-11eb-9c22-b2bace10ed99.png">
테스트 파일 추가

yj-leee commented:
<img width="1140" alt="스크린샷 2020-11-23 오후 11 51 21" src="https://user-images.githubusercontent.com/70866485/99978938-e8b8ff00-2de9-11eb-947f-f09b25c7deee.png">

yj-leee commented:
![Uploading 스크린샷 2020-11-23 오후 11.51.35.png…]()

Commit:
MODIFY: analysis views.py 수정

1. 데코레이터 파일 request.user 수정

2. urls.py, views.py Class name 수정

3. view 기능
- movie 상세 페이지의 별점 분포(해당 movie 별점을 체크한 전체 user기준)
- user 마이 페이지의 별점 분포도 (user개인이 체크한 모든 영화에 대한 별점 분포)
- movie_id를 받으면 movie에 대한 별점을 리턴
- movie_id를 받지 않으면 user의 개인 별점 분포 리턴

Commit:
MODIFY: analysis views.py 수정

1. 로그인 데코레이터 user 수정
2. analysis views.py 수정

Commit:
conflict 해결
Merge branch 'main' of https://github.com/wecode-bootcamp-korea/14-1st-Gotchapedia-backend into feature/star

devops-andrew commented:
LGTM
